code/modules/mob/living/carbon/carbon.dm

```diff
@@ -358,17 +358,32 @@
 		H.play_xylophone()
 
 /mob/living/carbon/flash_eyes(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, laser_pointer = FALSE, type = /obj/screen/fullscreen/flash)
+
+	var/obj/item/organ/internal/eyes/E = get_int_organ(/obj/item/organ/internal/eyes)
+
+	//Checks that should stop flashing entirely
+	if(ishuman(src))
+		var/mob/living/carbon/human/H = src
+		if((!E && !(H.dna.species.bodyflags & NO_EYES)))
+			return
+
+	if(E && E.status & ORGAN_DEAD)
+		return
```

Comment:
Can you make a `can_get_flashed` proc which you override to add additional checks?
This way you can have the `can_get_flashed` be called in the base of `flash_eyes` and handle it like that. Regular living mobs would just have `can_get_flashed` do the check located in `/mob/living/proc/flash_eyes`, carbons would have the eye existence + alive eyes check. And the human override would finally add the dna.species check you added.

This also allows you to still send the signal in the existing logic

Reply:
So replace all the if -> return checks with a new proc separate to `flash_eyes` that's defined in `living.dm` with the checks done there (for traits and stuff), then overridden in `carbon.dm` with the checks for eyes, and overridden again in `human.dm` with the species-based check? Then, in `flash_eyes`, I do `if(!can_get_flashed(src)) return` (or `if(flashing_blocked(src)) return`) instead of all the other stuff?

Reply:
Only the base proc should do the check once. Then return FALSE if the mob can't be flashed. Then the child procs (carbon/flash_eyes) will check the return value of the parent call. False? can't be flashed.

Reply:
I think I get it, because of the `if(.)` in `carbon/flash_eyes`, right?

Reply:
Yep yep!